### PR TITLE
handle non-breaking spaces for indentation indication

### PR DESCRIPTION
**Pull request by ahicks92**
I've seen this a lot, only just figured out what it is.  if you're in the web browser and indentation indication is on, non-breaking spaces cause it to blow up annoyingly.  For an example, see [this thread](http://forum.audiogames.net/viewtopic.php?pid=242924#p242924), post 6.  It causes indentation to be read in Firefox, Thunderbird, etc. as "space space space space..."
If someone says where the indentation function is, I can probably submit a PR.  I imagine this isn't hard.

### CODE REVIEW

**Comment by camlorn** [comment]
This works for the first part.  I left it open because I'm not sure if we want to do the second or not.
Sorry about the history, I'm not sure why the pull request is including both commits.  I used [hub](https://github.com/github/hub) and it says that pull-request to issue conversion is deprecated...but only after it does it.  I'm assuming this is the bug, and suffice it to say I will not be doing that again.

File: source/speech.py [hunk 1]
```diff
@@ -421,6 +422,8 @@ def getIndentationSpeech(indentation):
 		# Translators: This is spoken when the given line has no indentation.
 		return _("no indent")
 
+	#The non-breaking space is semantically a space, so we replace it here.
+	indentation = RE_NBSP_CONVERT.sub(" ", indentation)
```
Comment (jcsteh) [inline source/speech.py]: Obviously totally cool with using regexps if it allows you to do multiple string passes in one step and thus improve performance, but in this case, is there any advantage of using a regexp over a simple:

```
indentation = indentation.replace(u"\xa0", " ")
```

**Comment by camlorn** [comment]
Yeah, actually. `"\xa0".replace(u"\xa0", " ")` fails because replace tries to decode the unicode literal to Ascii.  Passing the ascii representation to the Unicode version of replace also breaks; in that case, I suspect that it's trying to decode the argument to ascii first, but it's hard to tell.  The function's docstring doesn't explicitly say that the input is going to be a Unicode string, so you either have to force it (create 2 temporary strings in quick succession) or do an even uglier if statement as far as I can see.  There might be a cleaner way, but a simple replace call seems to break on one of the alternatives no matter how I write it, the if statement switching on types seemed ugly, and you seemed concerned that this be absolutely as fast as possible.
I don't  discount the possibility that I missed something, me and Unicode do not mix.

**Comment by jcsteh** [comment]
ah, very good point. However, it seems this fails already for Unicode:

```
>>> speech.getIndentationSpeech(" \xa0")
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "C:\Users\jamie\src\nvda\source\speech.py", line 438, in getIndentationSpeech
    return " ".join(res)
UnicodeDecodeError: 'ascii' codec can't decode byte 0xa0 in position 0: ordinal not in range(128)
```

Given this and the fact that I can't think of a case when it would ever be passed an ANSI string, just change the docstring to say unicode instead of basestring for the parameter type, which should also make the unicode.replace safe.

**Comment by camlorn** [comment]
That should do it.

**Comment by jcsteh** [comment]
I didn't see this before, but I see you've based this on (and are requesting merge into) next. For future reference, all pull requests should be based on (and submitted against) master; see the [Contributing guidelines](https://github.com/nvaccess/nvda/wiki/Contributing). No need to resubmit or anything for this one; I'll take care of it.

**Comment by nvaccessAuto** [comment]
Incubated in 67f080dd895c60d4fe33604aa9d9ae5cc7b08e2a.

**Pull request closed without merging**